dlc-manager/src/manager.rs

```diff
@@ -1941,14 +1953,39 @@ where
         Ok(())
     }
 
-    fn on_reject(&mut self, reject: &Reject, counter_party: &PublicKey) -> Result<(), Error> {
-        let mut signed_channel =
-            get_channel_in_state!(self, &reject.channel_id, Signed, Some(*counter_party))?;
+    fn on_reject(&self, reject: &Reject, counter_party: &PublicKey) -> Result<(), Error> {
+        let channel = self.store.get_channel(&reject.channel_id)?;
+
+        if let Some(channel) = channel {
+            if channel.get_counter_party_id() != *counter_party {
+                return Err(Error::InvalidParameters(format!(
+                    "Peer {:02x?} is not involved with {} {:02x?}.",
+                    counter_party,
+                    stringify!(Channel),
+                    channel.get_id()
+                )));
+            }
+            match channel {
+                Channel::Offered(offered_channel) => {
```

Comment:
I think we'd need to unlock the utxos here, but it's a bit a big change so let me know if you want to do that here or just add an issue and someone can do it later.

Reply:
Good catch! I will have a look.

Reply:
Addressed in 19b0e87cc39386f2e0fd041002d904e4cdd64ea0